Migrate project (branch RoleManager) net5.0 to net8.0

AngeloDotNet opened the pull request.

AngeloDotNet committed:
Refactoring InputModels

AngeloDotNet committed:
Refactoring ViewModels

AngeloDotNet committed:
Refactoring Enums

AngeloDotNet committed:
Refactoring Entities

AngeloDotNet committed:
Refactoring models Auth

AngeloDotNet committed:
Refactoring DBContext

AngeloDotNet committed:
Update README

AngeloDotNet committed:
Refactoring and update solutions files

BrightSoul commented:
Ciao e grazie per la PR! Occhio, c'è il tuo nome e indirizzo email nel db, meglio rimuoverlo.
Casomai elimina e rigenera il db usando le migration, poi farò il merge.

AngeloDotNet committed:
Fix README

AngeloDotNet committed:
Pulizia dati database

AngeloDotNet commented:
Fixato il nome corretto (ASP.NET Core 8) in questo commit: https://github.com/BrightSoul/UserManagementUiDemo/pull/2/commits/6a2297c9a785180e3e891e9035211799609d2cd3 ed effettuata pulizia del database.

AngeloDotNet commented:
This pull request includes several changes to modernize the codebase by adopting file-scoped namespaces and updating the target framework. The most important changes include refactoring namespaces, updating the target framework, and cleaning up the code formatting.

### Refactoring namespaces:

* Converted to file-scoped namespaces across multiple files for cleaner and more concise code. (`Data/ApplicationDbContext.cs`, `Models/Authorization/PermissionAuthorizationHandler.cs`, `Models/Authorization/PermissionPolicyProvider.cs`, `Models/Authorization/PermissionRequirement.cs`, `Models/Entities/ApplicationRole.cs`, `Models/Entities/ApplicationUser.cs`, `Models/Enums/Permission.cs`, `Models/InputModels/EditClaimInputModel.cs`, `Models/InputModels/EditRoleInputModel.cs`, `Models/InputModels/RoleCreateInputModel.cs`, `Models/InputModels/RoleEditInfoInputModel.cs`, `Models/InputModels/UserCreateInputModel.cs`, `Models/InputModels/UserEditProfileInputModel.cs`, `Models/ViewModels/UserListViewModel.cs`, `Program.cs`, `Startup.cs`) [[1]](diffhunk://#diff-73c857fa5717f30789b1fab27594b694d56f01e563c18cc1123846a894a93798L6-R19) [[2]](diffhunk://#diff-bb02663b629b8e16f49ac74a92c27eccd1df273b84085fcbdb19b24c0e50f079L1-R6) [[3]](diffhunk://#diff-3cdf606e92d0b7bb7ea6fc06b352aa960214beda476b3ad021a880de814ae989L8-R15) [[4]](diffhunk://#diff-6513f2bb907223fd55712060b2299bd69e9776d7dfe0279c11ea4ae81f10192cL3-R8) [[5]](diffhunk://#diff-fe027fcdb872d8ff5fe22fc3227792598604b9c13d415a50a90cb43979f3209fL4-L12) [[6]](diffhunk://#diff-1f3bd231b8d02b3c050170b1c432f4ca46abd818b98b1c55da1d33b9f5be4763L4-L12) [[7]](diffhunk://#diff-ba10184d24699a6325e6b42d58bd5e8002f530025bd4fbbc0a3e59dca8f52e76L1-L10) [[8]](diffhunk://#diff-c70a1ab687db13c2e3adef46125db790c87b15bc1e185e01d6dbc0c5f390b4a3L1-R5) [[9]](diffhunk://#diff-0198404cf9fed846205e089a9a0d18da79b83710d5b56207da20cffa620ca725L6-R7) [[10]](diffhunk://#diff-c5f811142245a550a03327ae0b4ecbf247b24d1de79b980b804cf8c137526785L5-R6) [[11]](diffhunk://#diff-d8005c60e6cc2ef1c79aafe9f8b6c559f9361de1622277b5d5b019fead6cd9f3L6-R7) [[12]](diffhunk://#diff-27c26312341c14ea7c50d870d8fb62ee62b6a3d896fe9e3fbc1aadc4f58a14f1L5-R6) [[13]](diffhunk://#diff-1c1e489bb1b22aa34f6a9fb21125ea06a113258130112429c837c1c1fd0b9d0bL6-R7) [[14]](diffhunk://#diff-402afcfd1c44aaca2c0f665be1f153656bc0bfa684aca0ce6e1242e411cb4cd1L1-R4) [[15]](diffhunk://#diff-0b69b473fe937040615d69f606751f61ddbc2e3a1849360ff2456c22afe88c0bL1-R5) [[16]](diffhunk://#diff-4e22c1d7ebc71b079da9964a9ac1a37440f83b247973de63c2c6bcdab18bd09dL19-R15)

### Updating target framework:

* Updated the target framework from .NET 5.0 to .NET 8.0 in the project file. (`UserManagementUiDemo.csproj`)
* Updated the README to reflect the new target framework version. (`README.md`)

### Code formatting improvements:

* Cleaned up and reformatted code for better readability and maintainability. (`Startup.cs`, `Program.cs`, `Models/InputModels/UserCreateInputModel.cs`) [[1]](diffhunk://#diff-4e22c1d7ebc71b079da9964a9ac1a37440f83b247973de63c2c6bcdab18bd09dL30-R32) [[2]](diffhunk://#diff-0b69b473fe937040615d69f606751f61ddbc2e3a1849360ff2456c22afe88c0bL26) [[3]](diffhunk://#diff-27c26312341c14ea7c50d870d8fb62ee62b6a3d896fe9e3fbc1aadc4f58a14f1L37-L52)
